### PR TITLE
Support API Resource (Collections)

### PR DESCRIPTION
**Pull request by lexdewilligen**
This pull request submits an alternative way of supporting API resources. We experienced a problem when passing API Resource Collections as props to Inertia. We didn't receive any meta for our collection and the collection itself wasn't wrapped in a data array. 

This is an alternative approach to #70. It's a fix for #27 and maybe also #28 (didn't test that out though).

### CODE REVIEW

**Comment by grantholle** [comment]
It would so great if this could be merged in. Is there something that needs to be done/approved? Over 3 weeks... 😕

**Comment by reinink** [comment]
@grantholle I work on this as I have time. Please be patient. 🙏

@lexdewilligen How is this approach different/better than the approach in #70?

**Comment by grantholle** [comment]
@reinink Yeah, of course! If you need help maintaining let me know.

**Comment by lexdewilligen** [comment]
@reinink I'm using [spatie's package](https://github.com/spatie/laravel-resource-links) to create resource links. However, in the current situation meta is not correctly being returned. This has something to do with the `toResponse()` not (correctly) being called if I remember it correctly. In the current situation, this is what Vue.js gets as prop:

```
organizations:Array[6]
    0:Object
    1:Object
    2:Object
    3:Object
    4:Object
    5:Object
```

However, I need the data to be wrapped in a data object alongside a meta object. Laravel creates this correctly. However, something goes wrong in the transfer between laravel -> inertia -> vue. 

The other solution provided still gives me this data in vue:

```
organizations:Array[6]
    0:Object
    1:Object
    2:Object
    3:Object
    4:Object
    5:Object
```

However, my solution provides the data in this way:
```
organization:Object
    data:Array[6]
        0:Object
        1:Object
        2:Object
        3:Object
        4:Object
        5:Object
    meta:Object
        can:Object
        links:Object
```
Which is what we're all trying to achieve.

**Comment by menthol** [comment]
This PR is better than #70

The resource should be detected juste before render, has the same level of callable detection.

One canvas : If we pull a resource as an entire response, this code do not work.

```
$books = BookResource::collection(Book::paginate());
return Inertia::render('Books/Index', $books)
```

File: src/Response.php [hunk 1]
```diff
@@ -52,14 +52,16 @@ public function toResponse($request)
...
+                ? Arr::only($this->props, $only)
+                : $this->props
+        )->map(function ($prop, $key) use ($request) {
+            if ($prop instanceof Closure) $prop = App::call($prop);
 
-        array_walk_recursive($props, function (&$prop) {
-            if ($prop instanceof Closure) {
-                $prop = App::call($prop);
-            }
+            return ($prop instanceof \Illuminate\Http\Resources\Json\ResourceCollection)
```
[the remark this answers is not in the document]
Comment (RSpeekenbrink) [reply src/Response.php]: ```Illuminate\Contracts\Support\Responsable``` sounds like the way to go in this case.

**Comment by reinink** [comment]
So, I've added support for "Responsable" props: 2b966ac27964534ee3bbe585b0ea0fff8babb53a

This approach is more like #70. I still don't understand how this was superior to that solution. Sorry if I've missed something.

I've tested this with both single resource responses and collection resource responses, and it works great.

**Pull request closed without merging**